llvm/lib/Target/AMDGPU/SIShrinkInstructions.cpp

```diff
@@ -550,15 +552,19 @@ bool SIShrinkInstructions::shrinkScalarLogicOp(MachineInstr &MI) const {
   uint32_t NewImm = 0;
 
   if (Opc == AMDGPU::S_AND_B32) {
-    if (isPowerOf2_32(~Imm)) {
+    if (isPowerOf2_32(~Imm) &&
+        MBB.computeRegisterLiveness(TRI, AMDGPU::SCC, std::next(MBBI), 2) ==
```

Comment:
I think it would be better to just check the `dead` flag on the SCC def.
Neighborhood search of 2 seems too small, `dead` flag is absolute.